compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -321,6 +330,47 @@ public void visit(BLangImportPackage importPkgNode) {
             return;
         }
 
+        // Detect cyclic module dependencies. This will not detect cycles which starts with the entry package because
+        // entry package has a version. So we check import cycles which starts with the entry package in next step.
+        if (importedPackages.contains(pkgId)) {
+            int index = importedPackages.indexOf(pkgId);
+            // Generate the import cycle.
+            StringBuilder stringBuilder = new StringBuilder();
```

Comment:
The code for import cycle generation is duplicated ne? Shall we extract that out as a separate method?

Reply:
This has been addressed with https://github.com/ballerina-platform/ballerina-lang/pull/15230.